classes/pdf/HTMLTemplateOrderSlip.php

```diff
@@ -29,12 +29,24 @@
  */
 class HTMLTemplateOrderSlipCore extends HTMLTemplateInvoice
 {
+    /**
+     * @var Order
+     */
     public $order;
+
+    /**
+     * @var OrderSlip
+     */
     public $order_slip;
 
+    /**
+     * @var int
+     */
+    private $id_cart;
+
     /**
      * @param OrderSlip $order_slip
-     * @param $smarty
+     * @param Smarty $smarty
```

Comment:
```suggestion
     * @param mixed $smarty
```

Reply:
It must be a Smarty instance to work

Reply:
You need to change the method signature too :)